secure-mobile-push/smp-android-client/app/src/main/java/de/adorsys/android/securemobilepushtest/MainActivity.java

```diff
@@ -0,0 +1,97 @@
+package de.adorsys.android.securemobilepushtest;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.provider.Settings;
+import android.support.annotation.NonNull;
+import android.support.v4.content.LocalBroadcastManager;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import de.adorsys.android.securemobilepush.sms.SmsReceiver;
+import de.adorsys.android.securemobilepush.sms.SmsTool;
+
+public class MainActivity extends AppCompatActivity {
+    @NonNull private TextView smsSenderTextView;
+    @NonNull private TextView smsMessageTextView;
+    @NonNull private LocalBroadcastManager localBroadcastManager;
+
+    private BroadcastReceiver broadcastReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (intent.getAction().equals(SmsReceiver.INTENT_FILTER_SMS)) {
+                String receivedTitle = intent.getStringExtra(SmsReceiver.KEY_SMS_SENDER);
+                String receivedMessage = intent.getStringExtra(SmsReceiver.KEY_SMS_MESSAGE);
+                smsSenderTextView.setText(getString(R.string.sms_sender_number, receivedTitle));
+                smsMessageTextView.setText(getString(R.string.sms_message, receivedMessage));
+            }
+        }
+    };
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           @NonNull String permissions[],
+                                           @NonNull int[] grantResults) {
+        switch (requestCode) {
+            case SmsTool.REQUEST_CODE_ASK_PERMISSIONS: {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length <= 0
+                        || grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+                    Toast.makeText(this, "App cannot work without RECEIVE_SMS permission!",
+                            Toast.LENGTH_LONG).show();
+                    startActivity(new Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS,
+                            Uri.parse("package:" + getApplicationContext().getPackageName())));
+                }
+            }
+        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        SmsTool smsTool = new SmsTool(this);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
method is annotated with
`@RequiresApi(api = Build.VERSION_CODES.M)`so the sdk version check should be redundant